File: 2483-minimum-penalty-for-a-shop/2483-minimum-penalty-for-a-shop.py
class Solution:
    def bestClosingTime(self, customers: str) -> int:
        res = 0
        y = 0
        n = 0
        yCur = 0
        nCur = 0
        for c in customers:
            if c == 'Y':
                y += 1
            else:
                n += 1
        pen = y
        for i,c in enumerate(customers):
            if c == 'Y':
                yCur += 1
            else:
                nCur += 1
            if (nCur + y - yCur) < pen:
                pen = nCur + y - yCur
                res = i + 1
        return res

File: 2483-minimum-penalty-for-a-shop/2483-minimum-penalty-for-a-shop.cpp
class Solution {
public:
    int bestClosingTime(string customers) {
        int res = 0;
        int pen = 0;
        int y = 0;
        int n = 0;
        int yCur = 0;
        int nCur = 0;
        for(int i = 0; i < customers.size(); i++)
        {
            if(customers[i] == 'Y')
            {
                y++;
            }
            else
            {
                n++;
            }
        }
        pen = y;
        for(int i = 0; i < customers.size(); i++)
        {
            if(customers[i] == 'Y')
            {
                yCur++;
            }
            else
            {
                nCur++;
            }
            if((nCur + y - yCur) < pen)
            {
                pen = nCur + y - yCur;
                res = i + 1;
            }
        }
        return res;
    }
};